http/src/handler.rs

```diff
@@ -352,7 +352,11 @@ impl<M: Metadata, S: Middleware<M>> RpcHandler<M, S> {
 		}
 
 		// Read metadata
-		let metadata = self.jsonrpc_handler.extractor.read_metadata(&request);
+		let handler = match self.jsonrpc_handler.upgrade() {
```

Comment:
Is it true that the only time we expect the upgrade to fail is when the tokio executor is closing?

Reply:
Yes, since that's the only place where we keep a strong reference.